tests/tests/oob_indexing.rs

```diff
@@ -230,3 +230,223 @@ impl TestResources {
         }
     }
 }
+
+#[gpu_test]
+static RESTRICT_DYNAMIC_BUFFERS: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(
+        TestParameters::default()
+            .downlevel_flags(wgpu::DownlevelFlags::COMPUTE_SHADERS)
+            .limits(wgpu::Limits::downlevel_defaults())
+            .skip(FailureCase::backend(Backends::all() - Backends::DX12)),
+    )
+    .run_async(|ctx| async move {
+        let shader_src = "
+            @group(0) @binding(0)
+            var<storage, read_write> in: u32;
+            @group(0) @binding(1)
+            var<storage, read_write> out: array<u32>;
+
+            struct T {
+                @size(16)
+                t: u32
+            }
+
+            @group(0) @binding(2)
+            var<uniform> in_data_uniform: array<T, 1>;
+
+            @group(0) @binding(3)
+            var<storage, read_write> in_data_storage: array<T, 1>;
+
+            @compute @workgroup_size(1)
+            fn main() {{
+                let i = in;
+                out[0] = in_data_uniform[i].t;   // should be 1 since we clamp the index
+
+                out[1] = in_data_storage[i].t;   // should be 3 since we rely on the D3D12 runtime to bound check and
+                                                 // the index is still in the bounds of the buffer
+
+                out[2] = in_data_storage[i+1].t; // should be 0 since we rely on the D3D12 runtime to bound check
```

Comment:
Could we write this such that it passes on every backend? If it's not possible, we should have the reasoning in a comment in the parameters.

Reply:
This test is quite specific to the way we bound check on the D3D12 backend. Vulkan has a set of allowed behaviors with `robust_buffer_access2`. On Metal we insert bound checks with the restrict setting by default, I guess that could be tested but the result in `out[1]` and `out[2]` will be `1` instead of `3` and `0`.

Should I update the test to include Metal support?

Reply:
Lets leave it the, but lets add a comment as to why this is DX12 only